network/transport/v2/handlers.go

```diff
@@ -300,6 +302,7 @@ func (p *protocol) handleGossip(ctx context.Context, connection grpc.Connection,
 	// If the XORs are not equal and the peer is behind, still request the missing refs if there are any.
 	tempXor := xor.Xor(refs...)
 	if tempXor.Equals(peerXor) || (msg.LC < clock && len(refs) > 0) {
+		p.state.CorrectStateDetected()
```

Comment:
add comment on why this is the correct state but the state on L266 is not. This is not trivial

Reply:
actually it should also have been on L266, just forgot it.